### PR TITLE
Playlist functionality

### PR DESCRIPTION
**Pull request by sushil-mod**
# Playlist Functionality 
Created modal for playlist added checkbox for adding video into a playlist for functionality created playlist slice which maintains the state of adding and removing playlist.

## Files to check
- PlaylistSlice.js
- modalSlice.js
- PlaylistCard.js
- playlistModal.jsx

### CODE REVIEW

File: src/components/card/VideoCard.jsx [hunk 1]
```diff
@@ -1,22 +1,81 @@
...
+//         if(!domNode.current.contains(e.target) ) {
+//           func();
+//         }}
+//       document.addEventListener("mousedown",handler);
+//     return () =>{
+//       document.removeEventListener("mousedown",handler);
+//     }
+//   })
+//   return domNode;
+// }
```
Comment (dev-enforced) [inline src/components/card/VideoCard.jsx:24]: Can shift this block of code to another utility if intending to use it in future or can remove it

File: src/components/card/VideoCard.jsx [hunk 2]
```diff
@@ -1,22 +1,81 @@
...
+//   })
+//   return domNode;
+// } 
 
 function VideoCard({video}) {
-  console.log(video)
+
+  const [showOption , setShowOption] = useState(false)
+  const { userToken } = useSelector((store)=>store.authentication)
+  // console.log(video)
```
Comment (dev-enforced) [inline src/components/card/VideoCard.jsx:30]: can remove this debugger

File: src/components/modal/PlaylistModal.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const { userToken } = useSelector((store)=>store.authentication)
+    const { playlists,selectedVideo } = useSelector((store)=>store.playlist)
+    
+    const dispatch = useDispatch();
+    const domNode = useClickOutside(()=>{dispatch(closeModal());dispatch(closeCreateModal());})
+    
+    console.log(userInput);
+    console.log(userToken);
+    console.log("playlists",playlists);
+    console.log("video",selectedVideo);
```
Comment (dev-enforced) [inline src/components/modal/PlaylistModal.jsx:23]: These debuggers shall also be removed

File: src/components/modal/PlaylistModal.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import React ,{ useState } from 'react'
+import {useSelector,useDispatch } from 'react-redux'
+import useClickOutside from '../../hooks/useClickOutside';
+import './PlaylistModal.css';
+import {useRef} from 'react';
```
Comment (dev-enforced) [inline src/components/modal/PlaylistModal.jsx:5]: Can shift it to L1 of this file

File: src/components/modal/PlaylistModal.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import { closeCreateModal, closeModal, openCreateModal } from '../../features/modal/modalSlice';
+import { addPlaylist, addVideoToPlaylist, removeVideoFromPlaylist } from '../../features/playlist/playlistSlice';
```
Comment (dev-enforced) [inline src/components/modal/PlaylistModal.jsx:7]: Can import all these utilities jointly from "../../features";

File: src/features/playlist/playlistSlice.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
+import {createAsyncThunk, createSlice } from '@reduxjs/toolkit';
```
Comment (dev-enforced) [inline src/features/playlist/playlistSlice.js:1]: Can start removing the debuggers in this file

**Review by dev-enforced — APPROVED** [review]
The live preview is working fine(https://deploy-preview-4--poetic-unicorn-6e22b8.netlify.app/) as mentioned with the features of creating new playlists as well as deleting them and then removing the existing ones and then using input checkboxes to add /remove the video from an existing  playlist. The playlist page with list of playlists is also working well!!
Code is also very well structured and easy to understand. Liked the use of a self made hook of useClickOutside for minimizing video options !!

**Review by abhijitdotsharma — APPROVED** [review]
- Love the text overlay on categories, but the white text on bottom left image is white, which is slightly uneasy to look at. You can fix that later on, not a major issue
- Inside `video-title-container` there is an h4 that is appearing, I think that can be improved later too.
- Playlist code looks good, and structured. Comments can be removed if not required for newer features

**Comment by sushil-mod** [comment]
> * Love the text overlay on categories, but the white text on bottom left image is white, which is slightly uneasy to look at. You can fix that later on, not a major issue
> * Inside `video-title-container` there is an h4 that is appearing, I think that can be improved later too.
> * Playlist code looks good, and structured. Comments can be removed if not required for newer features

will change the color and improve it ,thanks for suggesting changes

File: src/features/playlist/playlistSlice.js (hunk 6 above)
Comment (sushil-mod) [reply src/features/playlist/playlistSlice.js:1]: yes, will remove in next pr

File: src/components/modal/PlaylistModal.jsx (hunk 5 above)
Comment (sushil-mod) [reply src/components/modal/PlaylistModal.jsx:7]: ok i will try this

**Pull request merged**